yt_dlp/extractor/huya.py

```diff
@@ -0,0 +1,131 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import hashlib
+import random
+from html import unescape
+
+from yt_dlp.compat import compat_urlparse, compat_b64decode
+
+from yt_dlp.utils import ExtractorError, try_get, str_or_none, int_or_none, js_to_json, update_url_query
+
+from yt_dlp.extractor.common import InfoExtractor
+
+
+class HuyaLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.|m\.)?huya\.com/(?P<id>[^/#?&]+)(?:\D|$)'
+    IE_NAME = 'huya:live'
+    IE_DESC = 'huya.com'
+    TESTS = [{
+        'url': 'https://www.huya.com/572329',
+        'info_dict': {
+            'id': '572329',
+            'title': 're:^.+$',
+            'description': 're:^.+$',
```

Comment:
use md5 if it's long

Reply:
In fact, this site supports live streaming, so the description and title may be changed every time a user goes live.

Reply:
```suggestion
            'title': str,
            'description': str,
```